raiden-ts/src/transport/epics.ts

```diff
@@ -195,16 +194,20 @@ export const initMatrixEpic = (
         );
       }
     }),
-    mergeMap(({ matrix, server, setup }) =>
+    withLatestFrom(config$),
+    mergeMap(([{ matrix, server, setup }, config]) =>
       // ensure displayName is set even on restarts
       from(matrix.setDisplayName(setup.displayName)).pipe(
-        // ensure we joined discovery room
         mergeMap(() =>
-          matrix.joinRoom(
-            `#raiden_${network.name || network.chainId}_discovery:${getServerName(server)}`,
+          merge(
+            // ensure we joined global rooms
+            ...globalRoomNames(config).map(globalRoom =>
+              from(matrix.joinRoom(`#${globalRoom}:${getServerName(server)}`)),
+            ),
           ),
         ),
-        mapTo({ matrix, server, setup }),
+        toArray(), // wait all promises to complete
```

Comment:
More of a question: I think we could simplify using [forkJoin](http://xgrommx.github.io/rx-book/content/observable/observable_methods/forkjoin.html)? Looks like the operator to be closest to `Promise.all()`. No need to change the code, it's just a question.

Reply:
forkJoin would work if I made it a higher order observable, an observable which emitted observables, and cared more about the final values. In this case, I just want to wait for everything to finish, and as toArray was already around, went for it. But yet, forkJoin would work as well